client/layout/masterbar/item.jsx

```diff
@@ -1,46 +1,49 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { Component } from 'react';
 import classNames from 'classnames';
 import noop from 'lodash/noop';
 import isFunction from 'lodash/isFunction';
 import Gridicon from 'gridicons';
 
-export default React.createClass( {
-	displayName: 'MasterbarItem',
-
-	propTypes: {
-		url: React.PropTypes.string,
-		onClick: React.PropTypes.func,
-		tooltip: React.PropTypes.string,
-		icon: React.PropTypes.string,
-		className: React.PropTypes.string,
-		isActive: React.PropTypes.bool,
-		preloadSection: React.PropTypes.func
-	},
-
-	_preloaded: false,
-
-	getDefaultProps() {
-		return {
-			icon: '',
-			onClick: noop
-		};
-	},
+class MasterbarItem extends Component {
+	constructor() {
+		super();
+		this._preloaded = false;
+		this.preload = this.preload.bind( this );
+	}
 
 	preload() {
 		if ( ! this._preloaded && isFunction( this.props.preloadSection ) ) {
 			this._preloaded = true;
 			this.props.preloadSection();
 		}
-	},
+	}
 
 	render() {
 		const itemClasses = classNames( 'masterbar__item', this.props.className, {
 			'is-active': this.props.isActive,
 		} );
 
+		if ( this.props.isNotesItem ) {
+			return (
+				<div
```

Comment:
Can we avoid the code duplication with this?
```
const Tag = this.props.isNotesItem ? 'div' : 'a';
return <Tag {...props}></Tag>;
```
The only difference between the two variants is the `href` prop, isn't it? We could issue a warning in the case when `isNotesItem && href`.